pkg/controller/controller_utils.go

```diff
@@ -757,8 +757,10 @@ func (s ActivePods) Less(i, j int) bool {
 	//       see https://github.com/kubernetes/kubernetes/issues/22065
 	// 4. Been ready for empty time < less time < more time
 	// If both pods are ready, the latest ready one is smaller
-	if podutil.IsPodReady(s[i]) && podutil.IsPodReady(s[j]) && !podReadyTime(s[i]).Equal(podReadyTime(s[j])) {
-		return afterOrZero(podReadyTime(s[i]), podReadyTime(s[j]))
+	readyTime1 := podReadyTime(s[i])
+	readyTime2 := podReadyTime(s[j])
+	if podutil.IsPodReady(s[i]) && podutil.IsPodReady(s[j]) && !readyTime1.Equal(readyTime2) {
+		return afterOrZero(readyTime1, readyTime2)
```

Comment:
The same optimization can be done in `(s ActivePodsWithRanks) Less` and in `(s ByLogging) Less`.

Reply:
@Miciah 
Please take another look.

Reply:
I was referring to the `podReadyTime` optimization: You could modify the other `Less` functions to compute `podReadyTime` only once for each pod.

Reply:
Addressed.